### ISSUE TITLE
DomainRedirects with env vars

### ISSUE THREAD

**Issue by mlecoq** [serverless-nextjs/serverless-next.js#838]
I have the following configuration

```
connect:
  component: '@sls-next/serverless-component@1.18.0'
  inputs:
    domain: ${env.DOMAIN}
    domainType: 'both'
    domainRedirects:
      www.${env.DOMAIN}: https://${env.DOMAIN}
    bucketRegion: ${env.APP_BUCKET_REGION}
```

### Describe the bug

I want to use env var to define redirection but it fails (it works with hardcoded values) 

### Actual behavior

The deployment fails (see logs)

### Expected behavior

I should have the same behavior with env vars

### Steps to reproduce

deploy a domain redirection with env vars

### Screenshots/Code/Logs

`Error: domainRedirects: domain.com is invalid. The URL is not in a valid URL format.
                                 at Builder.normalizeDomainRedirects (/root/.serverless/components/registry/npm/@sls-next/serverless-component@1.18.0/node_modules/@sls-next/lambda-at-edge/dist/build.js:435:23)`

### Versions

- OS/Environment: Linux
- @sls-next/serverless-component version: 1.8.0
- Next.js version: 10

### Checklist

- [ X ] You have reviewed the README and [FAQs](https://github.com/serverless-nextjs/serverless-next.js#faq), which answers several common questions.
- [ ] Please first try using the most recent `latest` or `alpha` `@sls-next/serverless-component` [release version](https://github.com/serverless-nextjs/serverless-next.js/releases), which may have already fixed your issue or implemented the feature you are trying to use. Note that the old `serverless-next.js` component and the `serverless-next.js` plugin are deprecated and no longer maintained.

**Comment by dphang** [comment]
The code tries to create a URL from the given value, if there is an exception then it's not a valid URL: https://github.com/serverless-nextjs/serverless-next.js/blob/f6c5156bf8bb4912528dfb3723227cca0790218a/packages/libs/lambda-at-edge/src/build.ts#L904

Is it because you are missing the protocol e.g `https://`. Make sure you are prepending the value with `https://`. Though from your configuration it has it, so not sure why it is failing...

**Comment by krikork** [comment]
I have the same issue on @1.19.0-alpha.21

Hardcoded value like this works:
```
    domainRedirects:
      sandbox.mydomain.io: https://www.sandbox.mydomain.io 
```

This fails:
```
    domainRedirects:
      "${env.DOMAIN}": https://www.${env.DOMAIN}
```

(I've tried without the quotations too)

Here's the log output:

```
Error: domainRedirects: sandbox.mydomain.io is invalid. The URL is not in a valid URL format.
                                 at Builder.normalizeDomainRedirects (/root/.serverless/components/registry/npm/@sls-next/serverless-component@1.19.0-alpha.21/node_modules/@sls-next/lambda-at-edge/dist/build.js:514:23)
                                 at Builder.<anonymous> (/root/.serverless/components/registry/npm/@sls-next/serverless-component@1.19.0-alpha.21/node_modules/@sls-next/lambda-at-edge/dist/build.js:238:18)
                                 at Generator.next (<anonymous>)
                                 at fulfilled (/root/.serverless/components/registry/npm/@sls-next/serverless-component@1.19.0-alpha.21/node_modules/@sls-next/lambda-at-edge/dist/build.js:5:58)
```

**Comment by dphang** [comment]
The error is the same, it is because `https://www.` is being removed when you add that way, maybe it is not the proper way to concatenate in `serverless.yml` configuration? I am not sure why `https://www.` is being removed. Could you please specify the full URL in the environment variable. For example, set `env.DOMAIN_REDIRECT` to `https://www.sandbox.mydomain.io`

**Comment by dphang** [comment]
Update: it seems it might be an issue with Serverless Framework? I did see a few very similate issues recently where if you have `${env.VARIABLE}-blah` it seems to ignore the concatenation and only take`${env.VARIABLE}`.

See for a workaround: https://github.com/serverless-nextjs/serverless-next.js/issues/530#issuecomment-680122810

It also doesn't work for me, my recommendation is don't use concatenated env variable or workaround above.

**Comment by mlecoq** [comment]
Thanks @dphang

**Comment by mtwalsh** [comment]
For anyone else running into this issue, I've tried the following:

```yaml
domain: ${env.DOMAIN}
subdomain: ${env.SUBDOMAIN}

myNextApplication:
  ...
  inputs:
    ...
    domain:
      - ${env.SUBDOMAIN}
      - ${env.DOMAIN}
    domainType: "both"
    domainRedirects:
      "${domain}": https://${subdomain}.${domain}
```
And also without the quotes, so `${domain}: https://${subdomain}.${domain}`.

So despite using the workaround for concatenating env vars: https://github.com/serverless-nextjs/serverless-next.js/issues/530#issuecomment-680122810, still no joy.

When I export the Lambda function I can see in the `manifest.json` both options (with/without quotes) result in:

`"domainRedirects":{"${domain}":"https://www.example.com"}`

So I assume that's the issue right there, it's not able to expand an environment variable when it's used as the YAML key? But the value does work just fine.

I've just switched to hardcoding, which works and is not really the end of the world either!
